pkg/actions/cluster/owned.go

```diff
@@ -0,0 +1,70 @@
+package cluster
+
+import (
+	"github.com/kris-nova/logger"
+	"github.com/pkg/errors"
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/cfn/manager"
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+	"github.com/weaveworks/eksctl/pkg/eks"
+	"github.com/weaveworks/eksctl/pkg/printers"
+)
+
+type OwnedCluster struct {
+	cfg          *api.ClusterConfig
+	ctl          *eks.ClusterProvider
+	stackManager *manager.StackCollection
+}
+
+func NewOwnedCluster(cfg *api.ClusterConfig, ctl *eks.ClusterProvider, stackManager *manager.StackCollection) (*OwnedCluster, error) {
+	return &OwnedCluster{
+		cfg:          cfg,
+		ctl:          ctl,
+		stackManager: stackManager,
+	}, nil
+}
+
+func (c *OwnedCluster) Upgrade(dryRun bool) error {
+	currentVersion := c.ctl.ControlPlaneVersion()
+	versionUpdateRequired, err := requiresVersionUpgrade(c.cfg.Metadata, currentVersion)
+	if err != nil {
+		return err
+	}
+
+	if err := c.ctl.LoadClusterVPC(c.cfg); err != nil {
+		return errors.Wrapf(err, "getting VPC configuration for cluster %q", c.cfg.Metadata.Name)
+	}
+
+	printer := printers.NewJSONPrinter()
+	if err := printer.LogObj(logger.Debug, "cfg.json = \\\n%s\n", c.cfg); err != nil {
+		return err
+	}
+
+	if versionUpdateRequired {
+		if err := updateVersion(dryRun, c.cfg, currentVersion, c.ctl); err != nil {
+			return err
+		}
+	}
```

Comment:
do we want to move this up to below line 29? (or move 29 down a bit if something is needed for `updateVersion`) looks weird to do the check so far after getting the value

Reply:
it sort of feels in the right place, we error out early if we can't check if theres an update, and then afterwards we load cluster information, log what we've found and then proceed with the upgrade.